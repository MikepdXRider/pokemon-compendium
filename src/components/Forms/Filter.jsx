import React from 'react'

export default function Filter({ typeDataArr, setSelectedTypeStr, selectedTypeStr }) {
    return (
        <label>
            Type
            <select value={selectedTypeStr} onChange={(e) => setSelectedTypeStr(e.target.value)}>
                <option value="All">All</option>
                {
                    typeDataArr.map(typeDataStr => <option value={typeDataStr} key={typeDataStr}>{typeDataStr}</option>)
                }
            </select>
        </label>
    )
}
