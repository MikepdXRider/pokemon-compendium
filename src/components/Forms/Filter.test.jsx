import { render } from "@testing-library/react";
import Filter from "./Filter.jsx";

const typeDataArr = ['test-type1', 'test-type2'];

it('matches snapshot', () => {
    const {container} = render(<Filter typeDataArr={typeDataArr}/> );
    expect(container).toMatchSnapshot();
})