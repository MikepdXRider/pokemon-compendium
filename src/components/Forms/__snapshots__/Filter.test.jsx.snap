// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <label>
    Type
    <select>
      <option
        value="All"
      >
        All
      </option>
      <option
        value="test-type1"
      >
        test-type1
      </option>
      <option
        value="test-type2"
      >
        test-type2
      </option>
    </select>
  </label>
</div>
`;
